app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityTest.kt

```diff
@@ -683,13 +754,82 @@ class NavigationDrawerActivityTest {
     }
   }
 
+  // TODO(#2535): Unable to open NavigationDrawer multiple times on Robolectric
+  @RunOn(TestPlatform.ESPRESSO)
+  @Test
+  fun testNavDrawer_openNavDrawer_debug_pressBack_configChange_homeIsSelected() {
+    launch<NavigationDrawerTestActivity>(
+      createNavigationDrawerActivityIntent(internalProfileId)
+    ).use {
+      it.openNavigationDrawer()
+      onView(withText(R.string.developer_options)).perform(click())
+      onView(isRoot()).perform(pressBack())
+      it.openNavigationDrawer()
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.fragment_drawer_nav_view))
+        .check(matches(checkNavigationViewItemStatus(NavigationDrawerItem.HOME)))
+    }
+  }
+
+  @Test
+  fun testNavDrawer_inDebugMode_openNavDrawer_devOptionsIsDisplayed() {
+    launch<NavigationDrawerTestActivity>(
+      createNavigationDrawerActivityIntent(internalProfileId)
+    ).use {
+      it.openNavigationDrawer()
+      onView(withId(R.id.developer_options_linear_layout)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testNavDrawer_inDebugMode_configChange_devOptionsIsDisplayed() {
+    launch<NavigationDrawerTestActivity>(
+      createNavigationDrawerActivityIntent(internalProfileId)
+    ).use {
+      it.openNavigationDrawer()
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.drawer_nested_scroll_view)).perform(swipeUp())
+      onView(withId(R.id.developer_options_linear_layout)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testNavDrawer_inDebugMode_devOptionsMenuItem_opensDeveloperOptionsActivity() {
+    launch<NavigationDrawerTestActivity>(
+      createNavigationDrawerActivityIntent(
+        internalProfileId
+      )
+    ).use {
+      it.openNavigationDrawer()
+      onView(withId(R.id.developer_options_linear_layout)).perform(nestedScrollTo())
+        .check(matches(isDisplayed())).perform(click())
+      intended(hasComponent(DeveloperOptionsActivity::class.java.name))
+      intended(hasExtra(DeveloperOptionsActivity.getIntentKey(), 0))
```

Comment:
```suggestion
      intended(hasExtra(DeveloperOptionsActivity.getIntentKey(), internalProfileId))
```

Reply:
Done. This test is now moved to `NavigationDrawerActivityDevTest.kt`. PTAL at https://github.com/oppia/oppia-android/blob/536f69fcc3f0866080ce71c27f1dfd9c92a0c8dc/app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityDevTest.kt#L289-L294
Thanks.